docs/guides/auth.md

```diff
@@ -0,0 +1,71 @@
+
+We don't need to worry about creating the request, we only need to extract the credentials from the `Authorization` header and verify them.
+!!! note
+
+    There are multiple ways to verify the credentials. Most involve checking the username in a database, then checking a hash of the password against the stored password hash for that username. This tutorial will not go over them 
+
+<br>
+
+To do this we first need to get the `Authorization` header as a string by using the following code:
+```cpp
+std::string myauth = req.get_header_value("Authorization");
+```
+<br>
+
+Next we need to isolate our encoded credentials and decode them as follows:
+```cpp
+std::string mycreds = myauth.substr(6);
+std::string d_mycreds = crow::utility::base64decode(mycreds, mycreds.size()/*, URLSafe? (true/false)*/);
+```
+<br>
+
+Now that we have our `username:password` string, we only need to separate it into 2 different strings and verify their validity:
+```cpp
+size_t found = d_mycreds.find(':');
+std::string username = d_mycreds.substr(0, found);
+std::string password = d_mycreds.substr(found+1);
+
+/*Verify validity of username and password here*/
+return true; //or false if the username/password are invalid
+```
+
+## Token Auth
+Tokens are some form of unique data that a server can provide to a client in order to verify the client's identity later. While on the surface level they don't provide more security than a strong password, they are designed to be less valuable by being *temporary* and providing *limited access*. Variables like expiration time and access scopes are heavily reliant on the implementation however.<br><br>
+
+### Access Tokens
+The kind of the token itself can vary depending on the implementation and project requirements: Many services use randomly generated strings as tokens. Then compare them against a database to retrieve the associated user data. Some services however prefer using data bearing tokens. One example of the latter kind is JWT, which uses JSON strings encoded in base64 and signed using a private key or an agreed upon secret. While this has the added hassle of signing the token to ensure that it's not been tampered with. It does allow for the client to issue tokens without ever needing to present a password or contact a server. The server would simply be able to verify the signature using the client's public key or secret.<br><br>
+
+### Using an Access Token
+Authenticating with an access token usually involves 2 stages: The first being scquiring the access token from an authority (either by providing credentials such as a username and a password to a server or generating a signed token). The scope of the token (what kind of information it can read or change) is usually defined in this step.<br><br>
```

Comment:
Typo
```suggestion
Authenticating with an access token usually involves 2 stages: The first being acquiring the access token from an authority (either by providing credentials such as a username and a password to a server or generating a signed token). The scope of the token (what kind of information it can read or change) is usually defined in this step.<br><br>
```